flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/OperatorCoordinatorCheckpointContext.java

```diff
@@ -27,15 +28,33 @@
  * This context is the interface through which the {@link CheckpointCoordinator} interacts with an
  * {@link OperatorCoordinator} during checkpointing and checkpoint restoring.
  */
-public interface OperatorCoordinatorCheckpointContext extends OperatorInfo {
+public interface OperatorCoordinatorCheckpointContext extends OperatorInfo, CheckpointListener {
 
 	void checkpointCoordinator(long checkpointId, CompletableFuture<byte[]> result) throws Exception;
 
 	void afterSourceBarrierInjection(long checkpointId);
 
 	void abortCurrentTriggering();
 
-	void checkpointComplete(long checkpointId);
+	/**
+	 * This method does not throw exception because the operator coordinator implementation
+	 * is supposed to call {@link OperatorCoordinator.Context#failJob(Throwable)} in case
+	 * of receiving an exception.
+	 *
+	 * @param checkpointId The ID of the checkpoint that has been aborted.
+	 */
+	@Override
+	void notifyCheckpointComplete(long checkpointId);
+
+	/**
+	 * This method does not throw exception because the operator coordinator implementation
+	 * is supposed to call {@link OperatorCoordinator.Context#failJob(Throwable)} in case
+	 * of receiving an exception.
+	 *
+	 * @param checkpointId The ID of the checkpoint that has been aborted.
+	 */
+	@Override
+	default void notifyCheckpointAborted(long checkpointId) {}
```

Comment:
why are these methods being overridden here?

Reply:
I guess my question is why are `notifyCheckpointComplete(long checkpointId)` and `notifyCheckpointAborted(long checkpointId)` overridden here since the extended interface (CheckpointListener) has the exact same methods. If you added a default empty implementation for notifyCheckpointComplete here it would make sense as that's extra behavior the base interface doesn't have. But that doesn't seem to be the case. Hence, my original question.

Reply:
Ah, now I see what you mean. The reason is that we don't want these two methods to throw checked exception. As explained in the Java doc, the implementation should invoke `OperatorCoordinator.Context#failJob()` to fail the job.